examples/Facemesh-parts/sketch.js

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2023 ml5
+//
+// This software is released under the MIT License.
+// https://opensource.org/licenses/MIT
+
+let facemesh;
+let video;
+let faces = [];
+let options = { maxFaces: 1, refineLandmarks: false, flipHorizontal: false };
+
+function preload() {
+  // Load the facemesh model
+  facemesh = ml5.facemesh(options);
+}
+
+function setup() {
+  createCanvas(640, 480);
+  // Create the webcam video and hide it
+  video = createCapture(VIDEO);
+  video.size(width, height);
+  video.hide();
+  // Start detecting faces from the webcam video
+  facemesh.detectStart(video, gotFaces);
+}
+
+function draw() {
+  // Draw the webcam video
+  image(video, 0, 0, width, height);
+
+  drawPartsKeypoints();
+  drawPartsBoundingBox();
+}
+
+// Draw keypoints for specific face element positions
+function drawPartsKeypoints() {
+  // If there is at least one face
+  if (faces.length > 0) {
+    for (let i = 0; i < faces[0].lips.length; i++) {
+      let lips = faces[0].lips[i];
+      fill(0, 255, 0);
+      circle(lips.x, lips.y, 5);
+    }
+  }
+}
+
+// Draw bounding box for specific face element positions
+function drawPartsBoundingBox() {
+  // If there is at least one face
+  if (faces.length > 0) {
+    let lipsX = [];
+    let lipsY = [];
+    for (let i = 0; i < faces[0].lips.length; i++) {
+      // Find the lips
+      let lips = faces[0].lips[i];
+      lipsX.push(lips.x);
+      lipsY.push(lips.y);
+    }
+    noFill();
+    rect(
+      min(lipsX),
+      min(lipsY),
+      max(lipsX) - min(lipsX),
+      max(lipsY) - min(lipsY)
+    );
+  }
```

Comment:
@B2xx @UpKindLikeWater The API design you selected has keys for different landmarks (`.lips.`) - but inside, you chose to keep the original structure of having arrays of points. This makes it very consistent with the "vanilla" model, but it also makes it a bit harder to use for simple tracking applications - since now it's up to the caller (user) to calculate e.g. the center point, or the extent of this landmark. (That's quite a bit of code above...)

Perhaps we could something like this instead:

```
[{
  lips: {
    cX: 300,   // center point calculated from all the points
    cY: 300,
    w: 10,     // width calculated from all the points
    h: 3,
    points: [  // actual points (if you want to draw them)
     { ... },
     { ... },
    ]
  },
  ...
}]
```

@shiffman Would this make sense to you also?

Reply:
@gohai yes, I like this idea! I think `cx` and `cy` might be a little confusing, if we just use `x` and `y` is that clear? Should we set them at the top left or the center?  I would also perhaps opt for `width` and `height` instead of `w` and `h`.

Reply:
We could also use `centerX` and `centerY`, but maybe `x` `y` `width` `height` that are aligned with the default "rectMode" makes the most sense?

Reply:
I am not very sure about this... having instant access to the center point could make a lot of fun examples even more concise. (e.g. drawing a circle over your nose, drawing a line between the eyes, calculating the distance between mouse and...) But I understand the call to not mixing rectModes in APIs.

How about we provide both? 🤔

```
  lips: {
    x: 250,   // top left
    y: 250,
    width: 100,
    height: 100,
    centerX: 300,
    centerY: 300,
    points: [  // actual points (if you want to draw them)
     { ... },
     { ... },
    ]
  },
```

Reply:
I'd like to provide both because I think it is useful for facemesh models that contains so many points in one face element. 

I think this structure will also affect the output of bodypose and handpose models, maybe @ziyuan-linn will have some thoughts as well?

Reply:
I am not too sure if bodypose and handpose have any bounding boxes. There might be one for the entire body and entire hand, which is less useful than the facemesh bounding boxes. I don't really have a strong preference for this. I can see some cases where `x, y, width, height` is more useful and other cases where `centerX, centerY` is more useful, so keeping both might be a good idea. 

An alternative to keeping both is to keep `x` and `y` at the top left corner by default and add a function `facemesh.rectMode()` which can switch `x` and `y` to center mode. However, this might become confusing and may introduce unnecessary complexity.

Reply:
Yes @ziyuan-linn I agree, I don't think we need to add this for handpose or bodypose, I think it's more specific to the way people may want to use facemesh and the complexity of so many keypoints.